src/debug/providerRegistry.ts

```diff
@@ -0,0 +1,41 @@
+import * as vscode from "vscode";
+
+import { IDebugProvider, IDockerParser } from "./debugInterfaces";
+import { JavaDebugProvider } from "./javaDebugProvider";
+import { JavaDockerResolver } from "./javaDockerResolver";
+
+const supportedProviders: IDebugProvider[] = [
+    new JavaDebugProvider(new JavaDockerResolver())

```

Comment:
I don't think the registry needs to know that the JavaDebugProvider uses the JavaDockerResolver - why can't that be an internal implementation detail of the JavaDebugProvider?